pkg/cli/admin/inspect/discovery.go

```diff
@@ -0,0 +1,56 @@
+package inspect
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/client-go/discovery"
+)
+
+func InspectDiscovery(ctx context.Context, destDir string, discoveryClient discovery.CachedDiscoveryInterface) error {
+	if err := writeDiscovery(ctx, filepath.Join(destDir, "discovery-apis.yaml"), discoveryClient, "/apis"); err != nil {
+		return err
+	}
+	if err := writeDiscovery(ctx, filepath.Join(destDir, "discovery-api.yaml"), discoveryClient, "/api"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeDiscovery(ctx context.Context, destFile string, discoveryClient discovery.CachedDiscoveryInterface, url string) error {
+	var responseContentType string
+	var statusCode int
+	apiBytes, err := discoveryClient.RESTClient().Get().
+		AbsPath(url).
+		SetHeader("Accept", discovery.AcceptV2).
+		Do(ctx).
+		ContentType(&responseContentType).
+		StatusCode(&statusCode).
+		Raw()
+	if err != nil {
+		return fmt.Errorf("discovery %q failed: %w", url, err)
+	}
+
+	if statusCode < 200 && statusCode > 299 {
+		return fmt.Errorf("discovery %q returned with status %v", url, statusCode)
+	}
```

Comment:
I think we can also ensure that the content type is discovery/v2 as in https://github.com/kubernetes/kubernetes/blob/462dd326c2e98d937a96d49002883000efe4b2d6/staging/src/k8s.io/client-go/discovery/discovery_client.go#L330 (in case library-go changes rely on discovery/v2)

```go
if isGVK, err := discovery.ContentTypeIsGVK(responseContentType, schema.GroupVersionKind{Group: "apidiscovery.k8s.io", Version: "v2", Kind: "APIGroupDiscoveryList"}); !isGVK || err != nil {
return fmt.Errorf("unexpected")
}
```

Reply:
> I think we can also ensure that the content type is discovery/v2 as in

good point.  will update